### DISCUSSION TITLE
Query About label shown in frontstages

### DISCUSSION THREAD

**Discussion by hsanand61564** [iTwin/itwinjs-core#1205] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/60655118/114867419-6b327180-9e12-11eb-8fee-e979cfec4819.png)

In the above viewer everywhere there is shown label like cube.left, cube.right, snapModeField.snapMode etc

Is there any global solution for the same or something that I may be missing.

**Comment by bsteinbk** [comment, marked as answer]
The text that is being displayed are keys to look up localized strings. Since the keys are displayed, this means that the app is either not delivering the localized strings, typically under the locales folder, or the localization is not being properly initialized. See https://github.com/imodeljs/itwin-viewer/blob/6a053db816563d7e48a71351f37a0a1effd6feb2/packages/modules/itwin-viewer-react/src/services/Initializer.ts#L245-L246 to see an example of how to initialize the localization. You can also review the network tab in the Chrome debugger to see if files like UiFramework.json are being requested and to see if the response contains actual JSON data.

**Reply by hsanand61564** [reply]
@bsteinbk When I tried to see the console, it showed like this

![image](https://user-images.githubusercontent.com/60655118/114871463-edbd3000-9e16-11eb-9a82-ca6c66436a98.png)


I believe I am missing something that is not giving any of the json files to the locales. Can u please help for the same

**Reply by aruniverse** [reply]
Please run a build of your app, webpack should copy and deliver the localization files. This is assuming youre using `bentley/react-scripts`

**Reply by calebmshafer** [reply]
As @aruniverse mentions, the `react-scripts build` script will copy all of the required localization files out of the dependencies for you.  We do have an open PR that will make it so that `react-scripts start`, and `npm run start`, handles that as well to avoid this type of issue.  Here's a link to that update so you can follow it, https://github.com/imodeljs/create-react-app/pull/37.
